### DISCUSSION TITLE
Questions(text) asked must appear in the StackOverflow title. Does not return any result if spelling mistake or even if some text matches with other questions.

### DISCUSSION THREAD

**Discussion by kushshiv** [IndianOpenSourceFoundation/dynamic-cli#147] [Show and tell]
# Title
Questions(text) asked by user must appear in the StackOverflow question title. Does not return any result in case of spelling mistake or even if some of the text matches with other questions.

# Explanation
Recently noted that if I searched for some question, and if there is a spelling mistake or if some of the text does not match the existing question in StackOverflow, then we don't get any matching query. 
It is because as per [documentation](https://api.stackexchange.com/docs/advanced-search), the entered text **must** match the question's title.

> title - text which must appear in returned questions' titles.

As a normal user I don't know what a StackOverflow question title must be having. This does not seems much relevant to me. 
Ideally it should return the most matched and most relevant text, as per my entered question i.e. even if some of the text in my question does not exist in the same question, it should show me at least if some text matched.
We can use `q`, to search for free form text parameter in place of `title`. Which says :

> q - a free form text parameter, will match all question properties based on an undocumented algorithm.

# Demo
I tried seaching for this question in StackOverflow:
https://stackoverflow.com/questions/70629789/env-data-not-accessible

But missed a spelling `accessible` to `accessable` and got not results.
![Screen Shot 2022-01-08 at 11 44 08 AM](https://user-images.githubusercontent.com/33164379/148634604-3f41876b-c03c-4dc0-b7da-a2527a00af65.png)

Even if the same question with spelling mistake searched in  StackOverflow, it gives some result and even gave result for the question I was searching for: 
![Screen Shot 2022-01-08 at 11 43 55 AM](https://user-images.githubusercontent.com/33164379/148634671-6cc3bed2-2158-4639-84e0-a9699fe4dbb3.png)

I think we have to change the search query and search based on `q` instead of `title`.

if I change from :
`return f"{self.search_content_url}/2.2/search/advanced?order=desc&sort=relevance&tagged={tags}&title={question}&site=stackoverflow"
`
to 
`return f"{self.search_content_url}/2.2/search/advanced?order=desc&sort=relevance&tagged={tags}&q={question}&site=stackoverflow"`

Its working properly and giving me relevant result.
![Screen Shot 2022-01-08 at 11 45 23 AM](https://user-images.githubusercontent.com/33164379/148634734-d0f0420b-b4d6-4221-83aa-0d9c065dbf90.png)

# Related Changes (PR): 
Changes mentioned here: https://github.com/IndianOpenSourceFoundation/dynamic-cli/pull/145

# Advise
What do you think @GouravSardana @pranavbaburaj

**Comment by lainq** [comment]
Thanks for finding the issue and letting us know 🙂 . As you have found the solution for it, I recommend you to create and pull request 😃 .

**Reply by kushshiv** [reply]
Hey @pranavbaburaj ... PR is already raised for this... Please check https://github.com/IndianOpenSourceFoundation/dynamic-cli/pull/145...

**Reply by lainq** [reply]
Oh sure
